documentation/Using-the-SDK/Creating-a-simple-sdk-plugin.md

```diff
@@ -155,47 +170,59 @@ The following are required for a Simple Data Source:
+4. Override the `GetDataSourceInfo` method. `DataSourceInfo` provides the driver some information about the data source
    to provide a better user experience. It is expected that this method will not be called before 
    `ProcessAsync`/`ProcessAsyncCore` because the data necessary to create a `DataSourceInfo` object might not be
    available beforehand.
 
-   ```
+   A `DataSourceInfo` contains three important pieces of information:
+   - How long, in nanoseconds, did the first event occur relative to the start of the sources being processed
+   - The duration, in nanoseconds, of the sources being processed
+   - The UTC wallclock time of the start of the sources being processed
+
+   ```cs
    public override DataSourceInfo GetDataSourceInfo()
    {
          return this.dataSourceInfo;   
    }
    ```
 
-5. Override the BuildTableCore method. This method is responsible for instantiating a given table. This method is called for each table identified by the SDK runtime as part of this plugin (see more on this in the 'Create Tables' section). The table to build is identified by a `TableDescriptor`, passed in as a parameter to this method. If the CDP isn't interested in the given table, it may return immediately.
+5. Override the `BuildTableCore` method. This method is responsible for instantiating a given table. This method is 
+called for each table identified by the SDK runtime as part of this CDS (see more on this in the "Create Tables" section). 
+The table to build is identified by the `TableDescriptor` passed in as a parameter to this method. If the CDP isn't interested 
+in the given table, it may return immediately.
 
-   ```
+   ```cs
    protected override void BuildTableCore(
       TableDescriptor tableDescriptor,
       Action<ITableBuilder, IDataExtensionRetrieval> buildTableAction,
       ITableBuilder tableBuilder)
    {
+      ...
    }
    ```
    
    How a table is created is left up to the plugin author. The sample for this documentation uses reflection to instantiate an instance of the table class.
 
-   After the table object is created, the `ITableBuilder` parameter is used to populate the table with columns and row data. In this example, the build logic exists in a `Build` method on the table, which is called by this method.
+   After the table object is created, the `ITableBuilder` parameter is used to populate the table with columns and row data. In this example, 
+   the build logic exists in a `Build` method on the table, which is called by this method.
 
-## Create tables
+## Create Tables
 
-Simple custom data sources provide tables as output. A table is set of columns and rows grouped together to provide output for related data. So tables must be created to have any output.
+Simple custom data sources provide tables as output. A table is set of columns and rows grouped together to provide output for related data. 
+Tables must be created (built) to be outputted by the plugin.
```

Comment:
emitted instead of outputted, is outputted a word?

Or you can change to "...to be a plugin output".

Reply:
Outputted is a word, and "emit" usually refers to streams of data, so I'd rather not say that. I like your second suggestion!